lib/list.js

```diff
@@ -93,8 +82,24 @@ util.inherits(List, Array);
 
 const _push = List.prototype.push;
 
+List.prototype.toItem = function(item) {
+  if (isClass(this.itemType)) {
+    return new this.itemType(item);
+  } else {
+    if (Array.isArray(item)) {
+      return item;
+    } else if (this.itemType === Date) {
+      if (item === null) return null;
+      return new Date(item);
+    } else {
+      return this.itemType(item);
+    }
+  }
+};
+
 List.prototype.push = function(obj) {
   const item = this.itemType && (obj instanceof this.itemType) ? obj : this.toItem(obj);
+  if (item && typeof item === 'object') applyParentProperty(item, this);
```

Comment:
replied to the discussion in https://github.com/strongloop/loopback-datasource-juggler/pull/1787/files#r363358834